uefi/src/proto/network/snp.rs

```diff
@@ -0,0 +1,587 @@
+            .into()
+        } else {
+            (self.receive_filters)(self, enable, disable, reset_mcast_filter, 0, None).into()
+        }
+    }
+
+    /// Modifies or resets the current station address, if supported
+    pub fn station_address(&self, reset: bool, new: Option<&MacAddress>) -> Result {
+        (self.station_address)(self, reset, new).into()
+    }
+
+    /// Resets statistics on a network interface
+    pub fn reset_statistics(&self) -> Result {
+        (self.statistics)(self, true, None, None).into()
+    }
+
+    /// Collects statistics on a network interface
+    pub fn collect_statistics(&self) -> Result<NetworkStats> {
+        let mut stats_table: NetworkStats = Default::default();
+        let mut stats_size = core::mem::size_of::<NetworkStats>();
+        let status = (self.statistics)(self, false, Some(&mut stats_size), Some(&mut stats_table));
+        Result::from(status)?;
+        Ok(stats_table)
+    }
+
+    /// Converts a multicast IP address to a multicast HW MAC Address
+    pub fn mcast_ip_to_mac(&self, ipv6: bool, ip: IpAddress) -> Result<MacAddress> {
+        let mut mac_address = MacAddress([0; 32]);
+        let status = (self.mcast_ip_to_mac)(self, ipv6, &ip, &mut mac_address);
+        Result::from(status)?;
+        Ok(mac_address)
+    }
+
+    /// Performs read operations on the NVRAM device attached to
+    /// a network interface
+    pub fn read_nv_data(&self, offset: usize, buffer_size: usize, buffer: *mut c_void) -> Result {
+        (self.nv_data)(self, true, offset, buffer_size, buffer).into()
+    }
+
+    /// Performs write operations on the NVRAM device attached to a network interface
+    pub fn write_nv_data(&self, offset: usize, buffer_size: usize, buffer: *mut c_void) -> Result {
+        (self.nv_data)(self, false, offset, buffer_size, buffer).into()
+    }
+
+    /// Reads the current interrupt status and recycled transmit buffer
+    /// status from a network interface
+    pub fn get_interrupt_status(&self) -> Result<InterruptStatus> {
+        let mut interrupt_status = InterruptStatus::new();
+        let status = (self.get_status)(self, Some(&mut interrupt_status), None);
+        Result::from(status)?;
+        Ok(interrupt_status)
+    }
+
+    /// Reads the current recycled transmit buffer status from a
+    /// network interface
+    pub fn get_recycled_transmit_buffer_status(&self) -> Result<Option<*mut u8>> {
+        let mut tx_buf: *mut c_void = ptr::null_mut();
+        let status = (self.get_status)(self, None, Some(&mut tx_buf));
+        Result::from(status)?;
+        if tx_buf.is_null() {
+            Ok(None)
+        } else {
+            Ok(Some(tx_buf.cast()))
+        }
+    }
+
+    /// Places a packet in the transmit queue of a network interface
+    pub fn transmit(
+        &self,
+        header_size: usize,
+        buffer: &[u8],
+        src_addr: Option<&MacAddress>,
+        dest_addr: Option<&MacAddress>,
+        protocol: Option<&u16>,
```

Comment:
For the public interface I don't think the `&` is needed for these three parameters. The types can be converted inside this function.